cmd/crank/beta/validate/manager.go

```diff
@@ -143,7 +146,32 @@ func (m *Manager) PrepExtensions(extensions []*unstructured.Unstructured) error
 				return errors.Wrapf(err, "cannot unmarshal configuration YAML")
 			}
 
-			m.confs[cfg.Name] = cfg
+			m.deps[cfg.Name] = cfg
+
+		case schema.GroupKind{Group: "meta.pkg.crossplane.io", Kind: "Provider"}:
+			meta, err := e.MarshalJSON()
+			if err != nil {
+				return errors.Wrap(err, "cannot marshal provider to JSON")
+			}
+			p := &metav1.Configuration{}
+			if err := yaml.Unmarshal(meta, p); err != nil {
+				return errors.Wrapf(err, "cannot unmarshal configuration YAML")
+			}
```

Comment:
```suggestion
			p := &metav1.Provider{}
			if err := yaml.Unmarshal(meta, p); err != nil {
				return errors.Wrapf(err, "cannot unmarshal provider YAML")
			}
```

Reply:
I wonder why we didn't catch this typo with the unit tests 🤔

Reply:
Oh that's totally my bad.. We couldn't catch this typo since a marshalled Provider can actually be unmarshalled to a `metav1.Configuration` due to their very similar fields. Since we unmarshal it after checking the kind we also couldnt see this typo with the help of its kind.